src/diffusers/pipelines/blip_diffusion/pipeline_blip_diffusion.py

```diff
@@ -0,0 +1,431 @@
+from dataclasses import dataclass
+from typing import List, Optional, Union
+
+import numpy as np
+import PIL
+from ...models import AutoencoderKL, UNet2DConditionModel
+from .modeling_ctx_clip import CtxCLIPTextModel
+from transformers import CLIPTokenizer
+from ...pipelines import DiffusionPipeline
+import torch
+from ...schedulers import PNDMScheduler
+from ...utils import (
+    BaseOutput,
+    is_accelerate_available,
+    is_accelerate_version,
+    logging,
+    randn_tensor,
+    replace_example_docstring,
+)
+
+from torch import nn
+from transformers.activations import QuickGELUActivation as QuickGELU
+from .modeling_blip2 import Blip2QFormerModel
+import tqdm
+from torchvision import transforms
+from torchvision.transforms.functional import InterpolationMode
+
+from PIL import Image
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+import re
+
+def numpy_to_pil(images):
+    """
+    Convert a numpy image or a batch of images to a PIL image.
+    """
+    if images.ndim == 3:
+        images = images[None, ...]
+    images = (images * 255).round().astype("uint8")
+    pil_images = [Image.fromarray(image) for image in images]
+
+    return pil_images
```

Comment:
No need to implement this here. We already have it: https://huggingface.co/docs/diffusers/main/en/api/utilities#diffusers.utils.numpy_to_pil